csharp/src/SeedLang.X/BaseParser.cs

```diff
@@ -0,0 +1,115 @@
+// Copyright 2021 The Aha001 Team.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Diagnostics;
+using Antlr4.Runtime;
+using Antlr4.Runtime.Tree;
+using SeedLang.Ast;
+using SeedLang.Common;
+
+namespace SeedLang.X {
+  // The abstract base class of the block parser and all SeedX language parsers.
+  //
+  // It provides interfaces to validate the SeedPython source code, and parse it into an AST tree
```

Comment:
"the SeedPython source code" - this class is not specific to SeedPython, isn't it?

Reply:
Done.